ADAL/src/ADAuthenticationParameters+Internal.m

```diff
@@ -38,8 +38,9 @@
 NSString* const UnauthorizedHTTStatusExpected = @"Unauthorized (401) HTTP status code is expected, but the actual status code is %d";
 const unichar Quote = '\"';
 //The regular expression that matches the Bearer contents:
-NSString* const RegularExpression = @"^Bearer\\s+([^,\\s=\"]+?)=\"([^\"]*?)\"\\s*(?:,\\s*([^,\\s=\"]+?)=\"([^\"]*?)\"\\s*)*$";
-NSString* const ExtractionExpression = @"\\s*([^,\\s=\"]+?)=\"([^\"]*?)\"";
+NSString* const BearerWithParamExpression = @"\\s*Bearer\\s+([^,\\s=\"]+?)\\s*=\\s*\"([^\"]*?)\"";
```

Comment:
My preference is for regex to be defined where it's used, and if a regex expression is used in more then one place that gets moved into a helper method.

Reply:
Removed all but ExtractionExpression. The code will be more complicated if we move it into a helper method.